packages/controller-utils/src/caip.ts

```diff
@@ -0,0 +1,75 @@
+import type { CaipChainId, Hex } from '@metamask/utils';
+import { parseCaipChainId, isCaipChainId } from '@metamask/utils';
+
+import { toHex } from './util';
+
+/**
+ * Checks whether the given value is a valid CAIP chain ID string for Ethereum.
+ *
+ * @param caipChainId - The CAIP chain ID to check for safety.
+ * @returns Whether the given CAIP chain ID is valid for Ethereum chains.
+ */
+export function isEthCaipChainId(
+  caipChainId: unknown,
+): caipChainId is CaipChainId {
+  if (!isCaipChainId(caipChainId)) {
+    return false;
+  }
+  const { namespace, reference } = parseCaipChainId(caipChainId);
+  const chainId = parseInt(reference, 10);
+  return namespace === 'eip155' && !Number.isNaN(chainId);
+}
+
+/**
+ * Creates a CaipChainId string that specifies an Ethereum chain.
+ *
+ * @param chainId - The chain ID in 0x prefixed hex or decimal string.
+ * @returns a valid CAIP chain ID for an Ethereum chain.
+ */
+export function buildEthCaipChainId(chainId = ''): CaipChainId {
+  const chainIdDecimal = chainId.startsWith('0x')
+    ? parseInt(chainId, 16).toString(10)
+    : chainId;
+
+  if (Number.isNaN(parseInt(chainIdDecimal, 10))) {
+    throw new Error(`Invalid chain ID "${chainId}"`);
+  }
+  return `eip155:${chainIdDecimal}` as CaipChainId;
```

Comment:
If passed a valid ethCaipChainId should we simply return the arg unmodified?

Reply:
yeah, makes sense to me